lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
 
     def get(self, request, number):
         """ HTTP handler. """
+        # If the account activation requirement is disabled for this installation, override the
+        # anonymous user object attached to the request with the actual user object (if it exists)
+        if not request.user.is_authenticated() and waffle.switch_is_active(DISABLE_ACCOUNT_ACTIVATION_REQUIREMENT_SWITCH):
+            try:
+                request.user = User.objects.get(id=request.session._session_cache['_auth_user_id'])
+            except DoesNotExist:
+                return JsonResponse(status=403)
```

Comment:
IIRC I was getting a 404 response from the LMS when the JS called back to the commerce api endpoint for order information